### PR TITLE
Scroll to Refresh

### PR DESCRIPTION
**Pull request by Perklone**

### CODE REVIEW

File: Sounds Like A Plan/Features/Profile-new/ProfileViewController.swift [hunk 1]
```diff
@@ -221,6 +232,19 @@ class ProfileViewController: UIViewController {
             fatalError()
         }
     }
+    
+    @objc func refreshEventData(_ sender: Any) {
+        if let userID {
+            dataRepository.fetchJoinedEvents(for: userID) { joinedEvent in
```
Comment (gremlinflat) [inline Sounds Like A Plan/Features/Profile-new/ProfileViewController.swift:238]: Do checking on segmented control first, then execute event fetching based on context (joined or hosted)

**Review by gremlinflat — CHANGES_REQUESTED** [review]
1. remove ProfileVC FETCHJOINEDEVENTS (line 110) and called a method that recently added

File: Sounds Like A Plan/Features/Home/HomeViewController.swift [hunk 2]
```diff
@@ -71,8 +72,18 @@ class HomeViewController: UIViewController, UISearchControllerDelegate {
         
         cellRegistration()
         view.addSubview(collectionView)
+        setupRefreshControl()
     }
     
+    private func setupRefreshControl() {
+        collectionView.addSubview(refreshControl)
+        refreshControl.addTarget(self, action: #selector(refreshEventData(_:)), for: .valueChanged)
```
Comment (gremlinflat) [inline Sounds Like A Plan/Features/Home/HomeViewController.swift:80]: didRefreshControlTriggered

File: Sounds Like A Plan/Features/Home/SeeAllViewController.swift [hunk 3]
```diff
@@ -55,9 +57,20 @@ class SeeAllViewController: UIViewController {
         collectionView.delegate = self
         
         cellRegistration()
+        setupRefreshControl()
         view.addSubview(collectionView)
     }
     
+    private func setupRefreshControl() {
+        collectionView.addSubview(refreshControl)
+        refreshControl.addTarget(self, action: #selector(refreshData(_:)), for: .valueChanged)
```
Comment (gremlinflat) [inline Sounds Like A Plan/Features/Home/SeeAllViewController.swift:66]: did___pastTense

**Comment by Perklone** [comment]
fixed #77  but, Apple Auth not working during testing

**Comment by Perklone** [comment]
image

**Comment by Perklone** [comment]
and just in case, xcodeprojnya jangan ikut diubah soalnya tadi gw ganti bundle identifiernya makanya berubah xcodeprojnya

**Comment by gremlinflat** [comment]
merged for further bug fixings

**Pull request merged**